### ISSUE TITLE
ERROR when runing Canu with Pacbio Hifi reads

### ISSUE THREAD

**Issue by Axolotl233** [marbl/canu#1907]
Hi
The program abort with ERRORs when I run `Canu 2.1.1` with a Pacbio hifi reads. There are message: 
```
ABORT: canu 2.1.1                                                                                                                                                       
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.                                                                                              
ABORT: Try restarting.  If that doesn't work, ask for help.                                                                                                             
ABORT:                                                                                                                                                                  
ABORT: Disk space available:  138483.082 GB                                                                                                                             
ABORT:                                                                                                                                                                  
ABORT: Last 50 lines of the relevant log file (unitigging/4-unitigger/unitigger.err):                                                                                   
ABORT:                                                                                                                                                                  
ABORT:                                                                                                                                                                  
ABORT:   Edge Confusion:                                                                                                                                                
ABORT:     Absolute              2500
ABORT:     Percent               15.0000
ABORT:
ABORT:   Unitig Construction:
ABORT:     Minimum intersection  500 bases
ABORT:     Maxiumum placements   2 positions
ABORT:
ABORT:   Debugging Enabled:
ABORT:     (none)
ABORT:                                                                                                                                                                  
ABORT:   ==> LOADING AND FILTERING OVERLAPS.                                                                                                                            
ABORT:                                                                                                                                                                  
ABORT:   ReadInfo()-- Found     1602388 reads.
ABORT:
ABORT:   OverlapCache()-- limited to 65536MB memory (user supplied).
ABORT:
ABORT:   OverlapCache()--     162MB for read data.
ABORT:   OverlapCache()--     649MB for best edges.
ABORT:   OverlapCache()--    2109MB for tigs.
ABORT:   OverlapCache()--     567MB for tigs - read layouts.
ABORT:   OverlapCache()--     811MB for tigs - error profiles.
ABORT:   OverlapCache()--   16384MB for tigs - error profile overlaps.
ABORT:   OverlapCache()--       0MB for other processes.
ABORT:   OverlapCache()-- ---------
ABORT:   OverlapCache()--   21089MB for data structures (sum of above).
ABORT:   OverlapCache()-- ---------
ABORT:   OverlapCache()--     405MB for overlap store structure.
ABORT:   OverlapCache()--   44040MB for overlap data.
ABORT:   OverlapCache()-- ---------
ABORT:   OverlapCache()--   65536MB allowed.
ABORT:   OverlapCache()--
ABORT:   OverlapCache()-- Retain at least 70 overlaps/read, based on 35.08x coverage.
ABORT:   OverlapCache()-- Initial guess at 135 overlaps/read.
ABORT:   OverlapCache()--
ABORT:   bogart: bogart/AS_BAT_OverlapCache.C:218: void OverlapCache::computeOverlapLimit(ovStore*, uint64): Assertion `totalOlaps > 0' failed.
ABORT:
ABORT:   Failed with 'Aborted'; backtrace (libbacktrace):
ABORT:   utility/src/utility/system-stackTrace.C::83 in _Z17AS_UTL_catchCrashiP7siginfoPv()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   bogart/AS_BAT_OverlapCache.C::218 in _ZN12OverlapCache19computeOverlapLimitEP7ovStorem()
ABORT:   bogart/AS_BAT_OverlapCache.C::155 in _ZN12OverlapCacheC2EPKcS1_djmm()
ABORT:   bogart/bogart.C::438 in main()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:
```
my comman line is 
```canu \         
-p asm -d SRR10948643 \
genomeSize=380m \
-pacbio-hifi ../data/SRR10948643.fasta \
useGrid=false maxThreads=30
```
i 'm a newble bioinfomatic, if you could kindly provide me some advice or point what is wrong with this, I will be appreciate.

**Comment by skoren** [comment]
The error message indicates you have no overlaps in this data, this doesn't seem like HiFi data. What does the rest of the report look like in terms of read lengths and histogram?

I took a quick look at the data on SRA and, even though the submitter marked it as CCS data, I don't think they are. Unfortunately since the original read IDs are lost by NCBI, it's hard to confirm this other than empirical analysis. First, the total data size is >200gb which is too large for a CCS run and doesn't match the 50x stated in the paper. Second, the reads have a very large SD in size, almost 5k. If you look at a typical CCS library, the SD is very small since the reads are tightly size selected. Third, I aligned the first 5k reads in this set to the reference used in the paper using minimap2 with the command `minimap2 -H -ax asm20 -r 5000`. Looking only at reads where their primary mapping covers >50% of the length, no reads align with over 92% identity and the mean identity (scaled by alignment length) is only 88%. 

In contrast if you look at a good HiFi rice library (https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR10238608) you see none of the above issues. The total data size is 25g, in line with expectation from a CCS library and consistent with about 50x coverage. Second, the SD is 4k (lower than the library you are using) but for a 20kb library. Last, the highest identity alignment to the same reference is 99.99% and the mean (scaled by alignment length) is 98.4% so more than 10% higher than your dataset even though this is not the same sample.

So in short, nothing you did and nothing in Canu, but an SRA/submission issue. I suggest contacting NCBI and/or the submitters to resolve.

**Comment by Axolotl233** [comment]
Hi,

This data was described of `https://doi.org/10.1101/2020.02.13.948489` , maybe something wrong with  SRA submission

Now i will try to run canu with data you recommended again, and you can close this issue temporary. If I meet other question, i will ask again (hope not~)

Thank you very much for your advice.

**Comment by skoren** [comment]
Yes, but as I said, the SRA submission doesn't actually match what's described the paper (500x vs 50x coverage for one) so I'd definitely report it to the authors and/or NCBI.
